internal/controlplane/handlers_repositories_test.go

```diff
@@ -470,8 +616,12 @@ func createServer(ctrl *gomock.Controller, repoServiceSetup repoMockBuilder, git
 				EncryptedToken: "encryptedToken",
 			}, nil).AnyTimes()
 		store.EXPECT().
-			GetProviderByID(gomock.Any(), gomock.Any()).
-			Return(provider, nil).AnyTimes()
+			ListRepositoriesByProjectID(gomock.Any(), gomock.Any()).
+			Return([]db.Repository{dbExistingRepo}, nil).AnyTimes()
+	}
+
+	if storeSetup != nil {
+		store = storeSetup(ctrl)
```

Comment:
I added this brutal override to avoid messing up other tests, but I believe it would be worth refactoring lines 589-through-621 completely. Let me know what you think.

Reply:
Maybe the `RepositoryService` needs a `ListRepositories` method. Then we wouldn't have to mock the store here. What do you think @dmjb?

Reply:
I agree. @blkt Let me know if you want to pair on this.

Reply:
I replaced direct access to store with access to repository service and removed all unnecessary utilities from tests.
I did not add tests for the Service itself, but the code path is still tested by tests on handler.